### DISCUSSION TITLE
<title>pushing .ipynb file from azure databricks to github

### DISCUSSION THREAD

**Discussion by ghost** [community/community#75591] [Other Feature Feedback, Questions, & Ideas]
### Azure databricks

Question

Hello,

Beginner here. 

I am pushing a .ipynb file from azure databricks to github repo, which gets converted to .py automatically. 
However, I need to push it as original format. 
Is there a way to achieve this ?

Thanks,
Chari

**Comment by Hsin-Hung** [comment]
Here is a tool to convert .py back to .ipynb:  https://github.com/Yoyodyne-Data-Science/ipynb-py-convert-databricks

**Reply by ghost** [reply]
Hi,
I need to push the file as .ipynb from databricks to github

**Reply by Hsin-Hung** [reply]
Databricks auto converts .ipynb to .py whenever you push since GitHub repo is mainly for source code, which explains the auto conversion. 

If you want to convert it back, you can use the tool or manually export the notebook as .ipynb and push. 

Hope it helps!

**Reply by ghost** [reply]
I also noticed that some notebooks which I pushed a month ago from DB have not converted to .py
Is this a new thing in DB ?

**Reply by Hsin-Hung** [reply]
No, it's not a new thing.
